### PR TITLE
✨ add query param to view any version of a resume

### PR DESCRIPTION
**Pull request by Ked57**
still working towards backing up resumes more easily. This PR aims to add the possibility to view older versions of a resume with query params. It works well

see #63 

image

Right now it just displays a blank page if there is no resume corresponding to the version_date entered

### CODE REVIEW

File: server/server.js [hunk 1]
```diff
@@ -138,6 +138,32 @@ function findByPath(req, res, path) {
...
+        last_modified
+      }
+    }
+    `,
+    { path: { _eq: path }, version_date: { _eq: version_date} },
+    req,
+    res,
+    function(data) {
+      if (data.rows.length < 1) {
+        findByUuid(req, res, path);
```
Comment (hgwood) [inline server/server.js:158]: Why is it falling back `findByUuid`? Shouldn't we respond 404 here?

File: server/server.js [hunk 2]
```diff
@@ -138,6 +138,32 @@ function findByPath(req, res, path) {
...
+      } else {
+        data.uuid = "";
```
Comment (hgwood) [inline server/server.js:160]: Why is this needed?

File: server/server.js [hunk 3]
```diff
@@ -166,7 +192,12 @@ function findByUuid(req, res, uuid) {
 // API
 api.get("/documents/:uuid", jwtCheck, (req, res) => {
   const uuid = req.params.uuid;
-  findByPath(req, res, uuid);
+  const { version_date } = req.query;
+  if (!version_date) {
+    findByPath(req, res, uuid);
+  } else {
+    findByPathForVersionDate(req, res, uuid, version_date);
+  }
```
Comment (hgwood) [inline server/server.js:200]: Let's keep var names in camel case :)
```suggestion
  const { version_date: versionDate } = req.query;
  if (!versionDate) {
    findByPath(req, res, uuid);
  } else {
    findByPathForVersionDate(req, res, uuid, versionDate);
  }
```

File: server/server.js (hunk 1 above)
Comment (Ked57) [reply server/server.js:158]: I just used code that was already in place for `findByUuid`

**Comment by hgwood** [comment]
The error case is a bit rough (URL changes to root but previously displayed resume is still displayed) but we can improve this later.
